application/src/test/java/org/opentripplanner/updater/trip/gtfs/moduletests/addition/AddedTest.java

```diff
@@ -123,14 +133,88 @@ void repeatedlyAddedTripWithNewRoute() {
     assertNotNull(env.getTransitService().getRoute(firstRoute.getId()));
   }
 
-  private TripPattern assertAddedTrip(String tripId, RealtimeTestEnvironment env) {
+  @Test
+  public void addedTripWithSkippedStop() {
+    var builder = new TripUpdateBuilder(
+      ADDED_TRIP_ID,
+      SERVICE_DATE,
+      ADDED,
+      TIME_ZONE,
+      "A loop",
+      "SW1234"
+    );
+    builder
+      .addStopTime(STOP_A_ID, "00:30", DropOffPickupType.PHONE_AGENCY)
+      .addSkippedStop(STOP_B_ID, "00:40", DropOffPickupType.COORDINATE_WITH_DRIVER)
+      .addSkippedStop(STOP_C_ID, "00:48")
+      .addStopTime(STOP_D_ID, "00:55", "A (non-stop)")
+      .addStopTime(STOP_A_ID, "01:00");
+    var tripUpdate = builder.build();
+
+    env.applyTripUpdate(tripUpdate);
+
+    // THEN
+    final TripPattern tripPattern = assertAddedTrip(ADDED_TRIP_ID, env);
+    assertEquals(PickDrop.CALL_AGENCY, tripPattern.getBoardType(0));
+    assertEquals(PickDrop.CANCELLED, tripPattern.getAlightType(1));
+    assertEquals(PickDrop.CANCELLED, tripPattern.getBoardType(1));
+    assertEquals(PickDrop.CANCELLED, tripPattern.getAlightType(2));
+    assertEquals(PickDrop.CANCELLED, tripPattern.getBoardType(2));
+    assertEquals(PickDrop.SCHEDULED, tripPattern.getAlightType(3));
```

Comment:
These assertions can be written more compactly with this notation: https://github.com/opentripplanner/OpenTripPlanner/blob/f0562318f5456dad90972e2b15e957c7af816931/application/src/test/java/org/opentripplanner/updater/trip/gtfs/moduletests/delay/DelayedTest.java#L64-L71

Reply:
I am testing the pattern, not the trip timetable here